### PR TITLE
WT-10526 Fix CMake build errors and docs on Window system

### PR DESCRIPTION
**Pull request by lukech**
Changes in this PR include:

- Fix to make Windows CMake builds compatible with systems that have unset `PROCESSOR_ARCHITECTURE` env variable
- Fix docs to git clone with https, and use "-A x64" to make builds work on 64-bit Windows systems. 

Changes has been verified on a Windows spawn host (distro `windows-64-vs2017-small`) with a successful CMake compilation and installation.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -21,7 +21,9 @@ include(cmake/define_libwiredtiger.cmake)
 # being defined (in base.cmake) since it will prevent it having a default value.
 if(NOT WT_ARCH)
     # Defer to our hosts architecture as our target architecture.
-    if ("${CMAKE_HOST_SYSTEM_PROCESSOR}" MATCHES "^(x86_64|i686|i386|AMD64)$")
+    # PROCESSOR_ARCHITECTURE env variable could be unset on certain Windows systems,
+    # in that case mapping the architecture to x86.
```
Comment (etienneptl) [inline CMakeLists.txt:25]: `mapping` -> `map`

**Review by etienneptl — APPROVED** [review]
LGTM, one typo

File: CMakeLists.txt (hunk 1 above)
Comment (lukech) [reply CMakeLists.txt]: Fixed in [d714263](https://github.com/wiredtiger/wiredtiger/pull/8772/commits/d714263002ab9093a48742333c6484821cfc5d7e)

**Comment by lukech** [comment]
evergreen merge

**Comment by lukech** [comment]
evergreen merge

**Pull request merged**